server/src/main/java/org/elasticsearch/indices/SystemIndices.java

```diff
@@ -334,18 +329,21 @@ public Predicate<String> getProductSystemIndexNamePredicate(ThreadContext thread
         return automaton::run;
     }
 
-    public Map<String, Feature> getFeatures() {
-        return featureDescriptors;
+    public Feature getFeature(String name) {
+        return getFeatures().stream().filter(f -> Objects.equals(name, f.getName())).findFirst().orElse(null);
```

Comment:
This turns looking up a feature by name into a linear operation. Is that ok? Seems like we should still perhaps have the map internally but not expose it? It would seem trivial to build the structure and keep it inside SystemIndices as a private member, but still expose the list of features elsewhere? Alternatively, we could keep the list, but sort it, and then use Collections.binarySearch. It won't be as fast as a hash lookup, but at least it would not be linear, and would probably be good enough for the limited use of looking up a specific feature.

Reply:
Right now I think there are fifteen or sixteen features at most, but of course features can be added by plugins, so I take your point. I'll go back to using a map internally.